docs/design/security.md

```diff
@@ -1,17 +1,106 @@
+   a.  If the user who started a long-lived process is removed from access to the cluster, the process should be able to continue without interruption
+   b.  If the user who started processes are removed from the cluster, administrators may wish to terminate their processes in bulk
+   c.  When containers run with a service account, the user that created / triggered the service account behavior must be associated to the container's action
+5. When container processes runs on the cluster, they should run in a [security context](https://github.com/GoogleCloudPlatform/kubernetes/pull/3910) that isolates those processes via Linux user security, user namespaces, and permissions.
+   a.  Administrators should be able to configure the cluster to automatically confine all container processes as a non-root, randomly assigned UID
+   b.  Administrators should be able to ensure that container processes within the same namespace are all assigned the same unix user UID
+   c.  Administrators should be able to limit which developers and project administrators have access to higher privilege actions
+   d.  Project administrators should be able to run pods within a namespace under different security contexts, and developers must be able to specify which of the available security contexts they may use
+   e.  Developers should be able to run their own images or images from the community and expect those images to run correctly
+   f.  Developers may need to ensure their images work within higher security requirements specified by administrators
+   g.  When available, Linux kernel user namespaces can be used to ensure b. and d. are met.
+   h.  When application developers want to share filesytem data via distributed filesystems, the Unix user ids on those filesystems must be consistent across different container processes
+6. Developers should be able to define [secrets](https://github.com/GoogleCloudPlatform/kubernetes/pull/2297) that are automatically added to the containers when pods are run
+   a.  Secrets are files injected into the container whose values should not be displayed within a pod. Examples:
+       1. An SSH private key for git cloning remote data
+       2. A client certificate for accessing a remote system
+       3. A private key and certificate for a web server
+       4. A .kubeconfig file with embedded cert / token data for accessing the Kubernetes master
+       5. A .dockercfg file for pulling images from a protected registry
+   b.  Developers should be able to define the pod spec so that a secret lands in a specific location
+   c.  Project administrators should be able to limit developers within a namespace from viewing or modify secrets (anyone who can launch an arbitrary pod can view secrets)
+   d.  Secrets are generally not copied from one namespace to another when a developer's application definitions are copied
```

Comment:
I hope we can also support the "dev and prod instances in the same namespace using labels for separation" model.  This means "Access to secrets can be controlled via the labels on a pod".